### ISSUE TITLE
In-Code Documentation with TSDoc

### ISSUE THREAD

**Issue by Hirschfuchs** [TypeStrong/ts-mockito#39]
As a developer, I would like to have an in-code-documentation, so that I can read the features and expected behaviours from within the IDE. I would strive for a gradual implementation, so that over the future development the documentation gets progressively better.

**Comment by roypeled** [comment]
I agree, we should try to add jsdoc annotations to the code, which most IDEs support. This will be a ongoing effort

**Comment by roypeled** [comment]
fixed in v3
